core/trino-main/src/main/java/io/trino/sql/planner/planprinter/DistributedFragmentRepresentation.java

```diff
@@ -0,0 +1,495 @@
+    private final Optional<WindowOperatorStatsRepresentation> windowOperatorStats;
+    private final NodeRepresentation nodeRepresentation;
+
+    @JsonCreator
+    public DistributedFragmentRepresentation(
+            @JsonProperty("name") String name,
+            @JsonProperty("identifier") String identifier,
+            @JsonProperty("layout") List<TypedSymbol> outputs,
+            @JsonProperty("details") String details,
+            @JsonProperty("cost") Optional<PlanNodeStatsAndCostSummary> planNodeStatsAndCostSummary,
+            @JsonProperty("estimates") List<EstimateStats> estimateStats,
+            @JsonProperty("fragmentStats") DistributedFragmentStats distributedFragmentStats,
+            @JsonProperty("children") List<DistributedFragmentRepresentation> children,
+            @JsonProperty("windowOperatorStats") Optional<WindowOperatorStatsRepresentation> windowOperatorStats,
+            @JsonProperty("nodeRepresentation") NodeRepresentation nodeRepresentation)
+    {
+        this.name = requireNonNull(name, "name is null");
+        this.identifier = requireNonNull(identifier, "identifier is null");
+        this.outputs = requireNonNull(outputs, "outputs is null");
+        this.details = requireNonNull(details, "details is null");
+        this.planNodeStatsAndCostSummary = planNodeStatsAndCostSummary;
+        this.estimateStats = estimateStats;
+        this.distributedFragmentStats = distributedFragmentStats;
+        this.children = children;
+        this.windowOperatorStats = windowOperatorStats;
+        this.nodeRepresentation = nodeRepresentation;
+    }
+
+    @JsonProperty
+    public String getName()
+    {
+        return name;
+    }
+
+    @JsonProperty
+    public String getIdentifier()
+    {
+        return identifier;
+    }
+
+    @JsonProperty
+    public List<TypedSymbol> getOutputs()
+    {
+        return outputs;
+    }
+
+    @JsonProperty
+    public String getDetails()
+    {
+        return details;
+    }
+
+    @JsonProperty
+    public Optional<PlanNodeStatsAndCostSummary> getPlanNodeStatsAndCostSummary()
+    {
+        return planNodeStatsAndCostSummary;
+    }
+
+    @JsonProperty
+    public List<EstimateStats> getEstimateStats()
+    {
+        return estimateStats;
+    }
+
+    @JsonProperty
+    public DistributedFragmentStats getDistributedPlanStats()
+    {
+        return distributedFragmentStats;
+    }
+
+    @JsonProperty
+    public List<DistributedFragmentRepresentation> getChildren()
+    {
+        return children;
+    }
+
+    @JsonProperty
+    public Optional<WindowOperatorStatsRepresentation> getWindowOperatorStats()
+    {
+        return windowOperatorStats;
+    }
+
+    @JsonIgnore
+    public NodeRepresentation getNodeRepresentation()
+    {
+        return nodeRepresentation;
+    }
+
+    public static DistributedFragmentRepresentation getPlanRepresentation(PlanRepresentation plan, NodeRepresentation node, Boolean verbose, Integer level)
```

Comment:
Some comments about this method:
* This method is misnamed. It returns a fragment representation, not a plan representation.
* What's the relationship between the `PlanRepresentation plan` argument and the `NodeRepresentation node` argument
* Use primitive types for `verbose` and `level`, instead of `Boolean` and `Integer`

Reply:
Renamed the method to `getFragmentRepresentation` and used primitive types.

PlanRepresentation is used to get the parent->child relationship between the nodes that are NodeRepresentation of all children for a given PlanNode. If NodeRepresentation contained children NodeRepresentations then PlanRepresentation wouldn't be needed but NodeRepresentation contains only PlanNodeIds as childrens. So, PlanRepresentation is used to get their NodeRepresentations.